gui/main.cxx

```diff
@@ -52,13 +56,42 @@ int main(int argc, char** argv)
   QApplication app(argc, argv);
   qtUtil::setApplicationIcon("mapgui");
 
+  bool help;
+  std::string openFilePath;
+
+  typedef kwiversys::CommandLineArguments argT;
+
+  argT arg;
+
+  arg.Initialize(argc, argv);
+
+  arg.AddBooleanArgument("--help", &help, "Display usage information");
+  arg.AddBooleanArgument("-h", &help, "Display usage information");
+  arg.AddArgument("--input", argT::SPACE_ARGUMENT,
+                  &openFilePath, "File to open at launch");
+  arg.AddArgument("-i", argT::SPACE_ARGUMENT,
```

Comment:
Since this is a Qt application, please use `qtCliArgs`. This interacts better with Qt-specific command line options that the application should continue to process (and to some extent, also better with Qt in general, e.g. the values you get back are `QString`, not `std::string`).

Example: https://github.com/Kitware/vivia/blob/master/Applications/VsPlay/main.cxx

How are you handling globs? I'd be... cautious... about relying on the shell because the order may not be as intended. OTOH, handling it in the app, while possible, means adding additional logic. (Another option might be to sort the list of files to be loaded.) Note also that shell handling requires handling multiple arguments. `qtCliArgs` does not support multi-argument options (only by repeating the option name), though if you're handling these as 'open anything supported', I'd recommend using positional arguments for that anyway...

The order of images with respect to other images is important, of course, but otherwise I didn't think it mattered?